common/numerated/src/tree.rs

```diff
@@ -427,131 +428,7 @@ impl<T: Numerated> IntervalsTree<T> {
-///
-/// See also [`IntervalsTree::and_not_iter`].
-pub struct AndNotIterator<
-    T: Numerated,
-    I1: Iterator<Item = NonEmptyInterval<T>>,
-    I2: Iterator<Item = NonEmptyInterval<T>>,
-> {
-    iter1: I1,
-    iter2: I2,
-    interval1: Option<NonEmptyInterval<T>>,
-    interval2: Option<NonEmptyInterval<T>>,
-}
-
-impl<
-        T: Numerated,
-        I1: Iterator<Item = NonEmptyInterval<T>>,
-        I2: Iterator<Item = NonEmptyInterval<T>>,
-    > Iterator for AndNotIterator<T, I1, I2>
-{
-    type Item = NonEmptyInterval<T>;
-
-    fn next(&mut self) -> Option<Self::Item> {
-        loop {
-            let interval1 = if let Some(interval1) = self.interval1 {
-                interval1
-            } else {
-                let interval1 = self.iter1.next()?;
-                self.interval1 = Some(interval1);
-                interval1
-            };
-
-            let interval2 = if let Some(interval2) = self.interval2 {
-                interval2
-            } else if let Some(interval2) = self.iter2.next() {
-                interval2
-            } else {
-                self.interval1 = None;
-                return Some(interval1);
-            };
-
-            if interval2.end() < interval1.start() {
-                self.interval2 = None;
-                continue;
-            }
-
-            self.interval2 = Some(interval2);
-
-            if interval1.end() < interval2.start() {
-                self.interval1 = None;
-                return Some(interval1);
-            } else {
-                if let Some(new_start) = interval2.end().inc_if_lt(interval1.end()) {
-                    self.interval1 = NonEmptyInterval::new(new_start, interval1.end());
-                    if self.interval1.is_none() {
-                        unreachable!("`T: Numerated` impl error: for each x: T, y: T, x < y ⇔ x.inc_if_lt(y) ≤ y");
-                    }
-                } else if interval1.end() == interval2.end() {
-                    self.interval1 = None;
-                    self.interval2 = None;
-                } else {
-                    self.interval1 = None;
-                }
-
-                if let Some(new_end) = interval2.start().dec_if_gt(interval1.start()) {
-                    let res = NonEmptyInterval::new(interval1.start(), new_end);
-                    if res.is_none() {
-                        unreachable!("`T: Numerated` impl error: for each x: T, y: T, x > y ⇔ x.dec_if_gt(y) ≥ y");
-                    }
-                    return res;
-                } else {
-                    continue;
-                }
-            }
-        }
-    }
-}
-
-impl<T: Numerated, D: Into<Interval<T>>> FromIterator<D> for IntervalsTree<T> {
+impl<T: Numerated, D: Into<IntervalIterator<T>>> FromIterator<D> for IntervalsTree<T> {
     fn from_iter<I: IntoIterator<Item = D>>(iter: I) -> Self {
         let mut tree = Self::new();
         for interval in iter {
```

Comment:
https://github.com/gear-tech/gear/blob/1128b5cc172e6fe004eded3a34d47782f304f849/common/numerated/src/tree.rs#L448
isn't it test for `.insert()` instead of `.try_insert()`?
I guess same for test `remove`

Reply:
https://github.com/gear-tech/gear/blob/1128b5cc172e6fe004eded3a34d47782f304f849/common/numerated/src/tree.rs#L625
it would be good to have more err-cases tests with asserting errors

Reply:
Agreed. I'd remove these `try_` methods at all because their semantic is misleading as insertion, removal (and so on) are actually infallible

Reply:
Ok, I'll thinking. May be better to remove them